web/src/invite.ts

```diff
@@ -308,6 +309,33 @@ function set_streams_to_join_list_visibility(): void {
     }
 }
 
+function update_guest_visible_users_count(): void {
+    const invite_as_value = $("#invite_as").val();
+
+    if (typeof invite_as_value !== "string") {
+        return;
+    }
+
+    const invite_as = Number.parseInt(invite_as_value, 10);
+    const guest_role_selected = invite_as === settings_config.user_role_values.guest.code;
+    if (!guest_role_selected || settings_data.guests_can_access_all_other_users()) {
+        $("#guest_visible_users_container").hide();
+        return;
+    }
+
+    const stream_ids = stream_pill.get_stream_ids(stream_pill_widget);
+    const visible_users_count = peer_data.count_guest_visible_users(stream_ids);
+
+    const message = $t(
+        {defaultMessage: "Guests will be able to see {N} users in their channels when they join."},
+        {N: visible_users_count},
+    );
+
+    const $help_link = $("#guest_help_link");
+    $("#guest_visible_users_message").text(`${message} `).append($help_link);
```

Comment:
Might be better to have a separate handlebars template for the text and link which accepts count of users as an argument. And just add that to the container before showing it.